File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/Issue.cs
using System;
using System.Collections.Generic;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class Issue : IAuditableEntity, IBaseEntity
    {
        public Issue()
        {
            Bill = new HashSet<Bill>();
            IssueLog = new HashSet<IssueLog>();
        }

        public int Id { get; set; }
        public string Summary { get; set; }
        public bool Warranty { get; set; }
        public DateTime? Deadline { get; set; }
        public int? StateId { get; set; }
        public int? AssignedToId { get; set; }
        public int VehicleId { get; set; }
        public int? MalfunctionId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }
        public int? Number { get; set; }
        public int Priority { get; set; }
        public DateTime Date { get; set; }

        public virtual Employee AssignedTo { get; set; }
        public virtual User Create { get; set; }
        public virtual Malfunction Malfunction { get; set; }
        public virtual User Mod { get; set; }
        public virtual State State { get; set; }
        public virtual Vehicle Vehicle { get; set; }
        public virtual ICollection<Bill> Bill { get; set; }
        public virtual ICollection<IssueLog> IssueLog { get; set; }
    }
}

File: CoreAngularDemo.Client/src/app/modules/shared/models/issue.ts
import { State } from './state';
import { Vehicle } from './vehicle';
import { Malfunction } from './malfunction';
import { TEntity } from '../../core/models/entity/entity';
import { Employee } from './employee';

export class Issue extends TEntity<Issue> {
  state: State;
  malfunction: Malfunction;
  warranty: boolean;
  vehicle: Vehicle;
  assignedTo: Employee;
  deadline: Date;
  summary: string;
  createdDate: Date;
  updatedDate: Date;
  priority: number;
  number: number;
  date: Date;

  constructor(issue: Partial<Issue>) {
    super(issue);
    this.state = new State(this.state);
    this.malfunction = new Malfunction(this.malfunction);
    this.vehicle = new Vehicle(this.vehicle);
    this.assignedTo = new Employee(this.assignedTo);
  }

  get deletable(): boolean {
    return this.state.name.toLowerCase() === 'new';
  }
}
